src/bin/pgcopydb/catalog.c

```diff
@@ -7700,6 +7704,169 @@ catalog_count_summary_done_fetch(SQLiteQuery *query)
 }
 
 
+/*
+ * catalog_add_timeline_history inserts a timeline history entry to our
+ * internal catalogs database.
+ */
+bool
+catalog_add_timeline_history(void *ctx, TimelineHistoryEntry *entry)
+{
+	DatabaseCatalog *catalog = (DatabaseCatalog *) ctx;
```

Comment:
Why use a `void *` context that actually is a `DatabaseCatalog *` argument? It would be better to skip the cast and have the compiler validation for us... is there a reason to do this that I can't see and that is better than having the compiler checks?

Reply:
I could not figure out the circular dependency in our headers. In the end I ended up writing some forward declaration for `DatabaseCatalog` to circumvent that.

Reply:
We might need a new header `pgsql_timeline.h` that could use both the catalogs and the lower-level pgsql header?

Reply:
I created a new header in a new commit, and removed the forward declaration for `DatabaseCatalog`

It contains the declaration of a function defined in `psql.c` just because I can not include the file that defines the struct `DatabaseCatalog` in `psql.h`.

```c
/* pgsql.c */
bool pgsql_start_replication(LogicalStreamClient *client, DatabaseCatalog *catalog,
							 char *cdcPathDir);
```